src/Juvix/Core/EAC/ConstraintGen.hs

```diff
@@ -12,7 +13,7 @@ import Prelude (error)
 {- Main functionality. -}
 
 -- Construct occurrence map.
-setOccurrenceMap ∷ (HasState "occurrenceMap" OccurrenceMap m) ⇒ Term → m ()
+setOccurrenceMap ∷ ∀ m primVal. (HasState "occurrenceMap" OccurrenceMap m) ⇒ Term primVal → m ()
```

Comment:
How vital are these foralls? are they needed now due to paramaterization?

Reply:
It was always forall, I just made it explicit.